cpp/harvest/marc_issn_lookup.cc

```diff
@@ -0,0 +1,190 @@
+        "- marc_output_articles is an output file generated by this tool.\n");
+
+    std::exit(EXIT_FAILURE);
+}
+struct SubFieldInfo {
+    std::string i_;
+    std::string t_;
+    std::string w_;
+    std::string x_;
+    int online_version_counter_;
+    int printed_version_counter_;
+    bool is_online_;
+    bool is_valid_;
+
+
+    void Builder(MARC::Record &record) {
+        for (auto &field : record) {
+            if (field.getTag() == "001")
+                w_ = "(DE-627)" + field.getContents();
+
+            if (field.getTag() == "245") {
+                MARC::Subfields subfields(field.getSubfields());
+                std::string subfield_a(subfields.getFirstSubfieldWithCode('a'));
+                std::string subfield_b(subfields.getFirstSubfieldWithCode('b'));
+                if (subfield_a != "" && subfield_b != "")
+                    t_ = subfield_a + " " + subfield_b;
+                else if (subfield_a != "" && subfield_b == "")
+                    t_ = subfield_a;
+                else if (subfield_a == "" && subfield_b != "")
+                    t_ = subfield_b;
+                else
+                    t_ = "";
+            }
+
+            if (field.getTag() == "300")
+                ((field.getFirstSubfieldWithCode('a') == "Online-Ressource") ? is_online_ = true : is_online_ = false);
+
+
+            if (field.getTag() == "773")
+                x_ = field.getFirstSubfieldWithCode('x');
+        }
+
+        i_ = "In:";
+    }
+};
+
+
+void UpdateSubfield(MARC::Subfields &subfields, const SubFieldInfo &sub_field_info) {
+    if (!subfields.replaceFirstSubfield('i', sub_field_info.i_))
+        subfields.addSubfield('i', sub_field_info.i_);
+    if (!subfields.replaceFirstSubfield('x', sub_field_info.x_))
+        subfields.addSubfield('x', sub_field_info.x_);
+    if (!subfields.replaceFirstSubfield('w', sub_field_info.w_))
+        subfields.addSubfield('w', sub_field_info.w_);
+
+    if (sub_field_info.t_ != "") {
+        if (!subfields.replaceFirstSubfield('t', sub_field_info.t_))
+            subfields.addSubfield('t', sub_field_info.t_);
+    }
+}
+
+void UpdateJournalValidity(std::map<std::string, SubFieldInfo> &journal_cache) {
+    // check whether the issn is occur more than once
+    for (auto sfi : journal_cache) {
+        if (sfi.second.online_version_counter_ == 1)
+            journal_cache[sfi.first].is_valid_ = true;
+        else if (sfi.second.online_version_counter_ > 1)
+            journal_cache[sfi.first].is_valid_ = false;
+        else if (sfi.second.online_version_counter_ == 0)
+            if (sfi.second.printed_version_counter_ == 1)
+                journal_cache[sfi.first].is_valid_ = true;
+            else
+                journal_cache[sfi.first].is_valid_ = false;
+    }
+}
+
+std::map<std::string, SubFieldInfo> JournalCacheBuilder(const std::string &input_journal_filename) {
```

Comment:
`BuildJournalCache` might be a better name, else we would assume its the constructor of a factory class.